src/Field.php

```diff
@@ -269,19 +275,9 @@ public function normalize($value)
             }
             break;
         case 'string':
-            // remove all line-ends and trim
-            $value = trim(str_replace(["\r", "\n"], '', $value));
-            if ($this->required && empty($value)) {
-                throw new ValidationException([$this->name => 'Must not be empty']);
-            }
-            break;
+            throw new Exception(['Use Field\ShortText for type=string', 'this'=>$this]);
         case 'text':
-            // normalize line-ends to LF and trim
-            $value = trim(str_replace(["\r\n", "\r"], "\n", $value));
-            if ($this->required && empty($value)) {
-                throw new ValidationException([$this->name => 'Must not be empty']);
-            }
-            break;
+            throw new Exception(['Use Field\Text for type=text', 'this'=>$this]);
```

Comment:
didnt' we agree on the meeting to keep backwards compatibility ?

if we accept that, it will break code in a very "python" way (no steady update path)

Reply:
It should be BC so far. Derived field classes don't call parent::normalize() so this switch is not used at all. Theoretically all this normalize method in Field class can be removed, but not 100% yet. That's why I left it there for cases when Field->type === null (some custom fields).